src/services/terra.py

```diff
@@ -46,18 +47,20 @@ def call_terra_api(self, sample_id, table):
                 'filterTerms': sample_id
             }
 
-            try:
-                response = requests.get(workspace_url, headers=headers, params=parameters)
-                response.raise_for_status()
-                data = response.json()
+            response = requests.get(workspace_url, headers=headers, params=parameters)
+            response.raise_for_status()
+            data = response.json()
 
-                if data.get('results'):
-                    results.extend(data['results'])
-                    page_number += 1
-                else:
-                    break
-            except requests.exceptions.RequestException as e:
-                print(f"Error calling Terra API: {e}")
-                break
+            if data.get('results'):
+                results.extend(data['results'])
+                page_number += 1
 
+                # Check if the next page exists based on metadata
+                metadata = data.get('resultMetadata', {})
+                filtered_page_count = metadata.get('filteredPageCount', 0)
+                print("page number", page_number)
+                print("filtered", filtered_page_count)
```

Comment:
Not sure if you still need these, or if they were just for testing? Might not really matter too much either way